`no-undefined-types`: bug with `@import`

## Expected behavior

If I import a type with the JSDoc `@import` and I use an property of this type, ESLint shouldn't report an error for the [`jsdoc/no-undefined-types`](https://github.com/gajus/eslint-plugin-jsdoc/blob/v51.2.1/docs/rules/no-undefined-types.md) rule.

## Actual behavior

ESLint reports error: _The type 'Foo.Bar' is undefined_

## Files

### `eslint.config.js`

```javascript
import jsdoc from "eslint-plugin-jsdoc";

export default [
  {
    plugins: { jsdoc },
    rules: {
      "jsdoc/no-undefined-types": "error",
    }
  }
];
```

### `type.d.ts`

```typescript
export namespace Foo {
    type Bar = string;
}
```

### `index.js`

```javascript
/**
 * @import { Foo } from "./type.d.ts"
 */

/**
 * @type {Foo.Bar}
 */
const baz = 42;
```

### `package.json`

```json
{
  "name": "testcase",
  "version": "1.0.0",
  "type": "module",
  "dependencies": {
    "eslint": "9.29.0",
    "eslint-plugin-jsdoc": "51.2.1",
    "typescript": "5.8.3"
  }
}
```

## To reproduce

1. `npm install`
2. `npx eslint`

   ```
   /home/regseb/testcase/index.js
     6:1  error  The type 'Foo.Bar' is undefined  jsdoc/no-undefined-types

   ✖ 1 problem (1 error, 0 warnings)
   ```

## Environment

- Node version: v22.15.0
- ESLint version: v9.29.0
- `eslint-plugin-jsdoc` version: 51.2.1

## Additional information

Type declaration and import is correct, because TypeScript finds the `Bar` property and reports type error:

1. `npx tsc --checkJs --noEmit index.js`

   ```
   index.js:8:7 - error TS2322: Type 'number' is not assignable to type 'string'.

   8 const baz = 42;
           ~~~


   Found 1 error in index.js:8
   ```

--- a/docs/rules/no-undefined-types.md
+++ b/docs/rules/no-undefined-types.md
@@ -943,5 +943,33 @@ export class Code {
     /** @type {AGlobal.AnotherMethod} */
     static #test5
 }
+
+import jsdoc from "eslint-plugin-jsdoc";
+
+/**
+ * @import { Linter } from "eslint"
+ */
+
+/**
+ * @type {Linter.Config}
+ */
+export default [
+  {
+    plugins: { jsdoc },
+    rules: {
+      "jsdoc/no-undefined-types": "error"
+    }
+  }
+];
+
+/**
+ * @typedef {object} Abc
+ * @property {string} def Some string
+ */
+
+/**
+ * @type {Abc['def']}
+ */
+export const a = 'someString';
 ````
 

--- a/src/rules/noUndefinedTypes.js
+++ b/src/rules/noUndefinedTypes.js
@@ -440,6 +440,8 @@ export default iterateJsdoc(({
           // Avoid appending for imports and globals since we don't want to
           //  check their properties which may or may not exist
           !imports.includes(val) && !globals.includes(val) &&
+          !importTags.includes(val) &&
+          !typedefDeclarations.includes(val) &&
           currNode && 'right' in currNode &&
           currNode.right?.type === 'JsdocTypeProperty') {
           val = val + '.' + currNode.right.value;

--- a/test/rules/assertions/noUndefinedTypes.js
+++ b/test/rules/assertions/noUndefinedTypes.js
@@ -1657,5 +1657,40 @@ export default /** @type {import('../index.js').TestCases} */ ({
         parser: typescriptEslintParser,
       },
     },
+    {
+      code: `
+        import jsdoc from "eslint-plugin-jsdoc";
+
+        /**
+         * @import { Linter } from "eslint"
+         */
+
+        /**
+         * @type {Linter.Config}
+         */
+        export default [
+          {
+            plugins: { jsdoc },
+            rules: {
+              "jsdoc/no-undefined-types": "error"
+            }
+          }
+        ];
+      `,
+    },
+    {
+      code: `
+        /**
+         * @typedef {object} Abc
+         * @property {string} def Some string
+         */
+
+        /**
+         * @type {Abc['def']}
+         */
+        export const a = 'someString';
+
+      `,
+    },
   ],
 });

